cmd/util/sign.go

```diff
@@ -12,92 +12,112 @@ package main
 import (
 	"crypto"
 	"crypto/ecdsa"
+	"crypto/ed25519"
 	"crypto/elliptic"
 	_ "crypto/sha256"
 	_ "crypto/sha512"
-	"encoding/asn1"
 	"fmt"
 	"io"
-	"math/big"
 )
 
-type ECDSASigner struct {
+// Signer represents an structure holding the signing information
+type Signer struct {
```

Comment:
This should be fine for now, but I'm guessing once we handle RSA we should probably have a separate struct for that.

Reply:
It could also be that we have a bool for each case: isRSA, is ed25519 and `curve` param for eddsa.